src/core/ColorPicker/ColorPicker.tsx

```diff
@@ -44,17 +44,74 @@ export const DefaultColors = [
 export type ColorPickerProps = {
   /**
    * Color Swatches that will be added within the color palette.
+   * Recommended to use `ColorSwatch` components.
    */
   children?: React.ReactNode;
+  /**
+   * Index of the selected color on the color palette.
+   */
+  activeColor?: number;
 } & Omit<CommonProps, 'title'>;
 
 export const ColorPicker = (props: ColorPickerProps) => {
-  const { children, className, ...rest } = props;
+  const { children, className, activeColor, ...rest } = props;
 
   useTheme();
 
+  const ref = React.useRef<HTMLDivElement>(null);
+  const [focusedColor, setFocusedColor] = React.useState(activeColor);
+
+  React.useEffect(() => {
+    setFocusedColor(activeColor);
+  }, [activeColor]);
+
+  const handleCalendarKeyDown = (
+    event: React.KeyboardEvent<HTMLDivElement>,
+  ) => {
+    const colorSwatches = ref.current?.querySelectorAll('.iui-color-swatch');
+    if (!colorSwatches?.length) {
+      return;
+    }
+
+    // If focusedColor is undefined, assume it is at index 0 and move from there
+    const currentIndex = focusedColor ?? 0;
+    let newIndex = -1;
+    switch (event.key) {
+      case 'ArrowDown':
+        newIndex = currentIndex + 10;
```

Comment:
will this really always be 10? The user might change the width or use a different swatch size.

So you might need to calculate this dynamically.

Reply:
Hmm I'm not sure how to do it dynamically, any ideas?